api/api.go

```diff
@@ -241,6 +242,17 @@ func defaultHttpClient() *http.Client {
 	return httpClient
 }
 
+func parseHeaders(headerString string) http.Header {
+	headers := http.Header{}
+	reader := bufio.NewReader(strings.NewReader("GET / HTTP/1.1\r\n" + headerString + "\r\n"))
+
+	logReq, err := http.ReadRequest(reader)
```

Comment:
It looks like we're constructing a request to extract the headers? Why not construct the headers directly?